docs/sources/administration/api-keys/index.md

```diff
@@ -44,28 +48,91 @@ This topic shows you how to create an API key using the Grafana UI. You can also
 
-### Before you begin
+> **Note:** If you are using API keys for authentication, we recommend that you migrate your integration to the service account authentication method. The API key will continue to work. You can locate the API key in the [Grafana service account tokens]({{< relref "../service-accounts/#service-account-tokens" >}}) details.
+
+This section shows you how to migrate your integration to use the new service account endpoints. You can migrate your API keys using:
+
+- The Grafana user interface
+- The Grafana API
+- Terraform
+
+### Migrate API keys to Grafana service accounts using the Grafana user interface
+
+This section shows you how to migrate API keys to Grafana service accounts using the Grafana user interface. You can choose to migrate a single API key or all API keys. When you migrate all API keys, you can no longer create API keys and must use service accounts instead.
 
-- Ensure you have permission to create Grafana service accounts. For more information about permissions, refer to [Roles and permissions]({{< relref "../roles-and-permissions/#" >}}).
+#### Before you begin
 
-**To migrate all API keys to service accounts:**
+To follow these instructions, you need:
+
+- To ensure you have permission to create Grafana service accounts. For more information about permissions, refer to [Roles and permissions]({{< relref "../roles-and-permissions/#" >}}).
+
+#### Steps
+
+To migrate all API keys to service accounts, complete the following steps:
 
 1. Sign in to Grafana, hover your cursor over **Configuration** (the gear icon), and click **API Keys**.
-2. In the top of the page, find the section which says **Switch from API keys to service accounts**
-3. Click **Migrate to service accounts now**.
-4. A confirmation window will appear, asking to confirm the migration. Click **Yes, migrate now** if you are willing to continue.
-5. Once migration is successful, you can choose to forever hide the API keys page. Click **Hide API keys page forever** if you want to do that.
+1. In the top of the page, find the section which says **Switch from API keys to service accounts**
+1. Click **Migrate to service accounts now**.
+1. A confirmation window will appear, asking to confirm the migration. Click **Yes, migrate now** if you are willing to continue.
+1. Once migration is successful, you can choose to forever hide the API keys page. Click **Hide API keys page forever** if you want to do that.
 
-**To migrate single API key to a service account:**
+To migrate a single API key to a service account, complete the following steps:
 
 1. Sign in to Grafana, hover your cursor over **Configuration** (the gear icon), and click **API Keys**.
 1. Find the API Key you want to migrate.
 1. Click **Migrate to service account**.
+
+### Migrate API keys to Grafana service accounts using the API
+
+This section shows you how to migrate API keys to Grafana service accounts using the Grafana API.
+
+#### Before you begin
```

Comment:
@eleijonmarck If there are any prerequisites or special permissions a user must have in place, we can add them to this Before you begin section. Are there any such prereqs? If not, we can remove this section.

Reply:
added permissions